src/main/java/com/xiaomi/infra/pegasus/rpc/Cluster.java

```diff
@@ -55,17 +58,25 @@ public static Cluster createCluster(Properties config) throws IllegalArgumentExc
         Integer.parseInt(
             config.getProperty(
                 PEGASUS_PUSH_COUNTER_INTERVAL_SECS_KEY, PEGASUS_PUSH_COUNTER_INTERVAL_SECS_DEF));
+
+    boolean enableWriteLimit =
```

Comment:
write-limit is not necessary to be initialized for each cluster, it can be initialized as a singleton. What about this way:

```java
class WriteLimitUtils 
{
  public static void validateSingleSet(...) throws IllegalArgumentException {
    if (isWriteLimitEnabled) {
      return;
    }
    ...
  }
  public static void validateMultiSet(...) throws IllegalArgumentException;
  public static void enableWriteLimit() {
    enableWriteLimit = true;
  }
  private static boolean enableWriteLimit = false;
}
```

Reply:
Well, it's better!